### ISSUE TITLE
Allow custom separators for action types

### ISSUE THREAD

**Issue by roboslone** [pburtchaell/redux-promise-middleware#165]
This is a feature request, so it doesn't fit any template =/

I use `redux-actions` with [nested structure](https://github.com/acdlite/redux-actions/issues/215) to create reducers, like so:
```js
const reducer = handleActions({
    search: {
        setQuery: (state, action) => { ... },
    },
    ...
})
```
With code above action with type `search/setQuery` will be processed by given reducer.
However, I cannot use such nested structure for actions dispatched by `redux-promise-middleware` since separator is [hardcoded](https://github.com/pburtchaell/redux-promise-middleware/blob/master/src/index.js#L80).

So I would like to change separator from `_` to `/` and use nested structure in `handleActions`:
```js
const reducer = handleActions({
    search: {
        fetchResults: {
            FULFILLED: (state, action) => { ... },
            PENDING: (state, action) => { ... },
            REJECTED: (state, action) => { ... },
        }
    },
    ...
})
```

**Comment by tomatau** [comment]
You can use type to reducer https://github.com/tomatau/type-to-reducer to handle nested actions. Your structure would suggest namespaces as well as separators e.g. `customer/bills/add_invoice` and that's tricky to configure, easier to use dynamic keys for the namespaces and nested action types for the separators

**Comment by roboslone** [comment]
`redux-actions` deals with namespaces, so `customer/bills/add_invoice` would be proceesed by following reducer:
```js
const reducer = handleActions({
    customer: {
        bills: {
            add_invoice: {
                PENDING: ...,
                ...
            },
        },
    },
})
```

**Comment by tomatau** [comment]
What's the benefit over using multiple reducers keyed by a namespace variable in conjunction with combineReducers from redux?

**Comment by roboslone** [comment]
I don't see how this is relevant to the issue.
I can't avoid hardcoded separator using multiple reducers and therefore I can't use `redux-actions` with `redux-promise-middleware` the way I like.

Perhaps I didn't understand you, could you please provide a code snippet?

Here's what I'm looking for:
```js
import { combineReducers } from 'redux'
import { createActions, handleActions } from 'redux-actions'


const actions = createActions({
    foo: {
        fetch: () => {
            return axios.get('/foo')
        }
    },
    bar: {
        fetch: () => {
            return axios.get('/bar')
        }
    }
})

const reducer = combineReducers({

    foo: handleActions({
        fetch: {
            PENDING: (state, action) => {...},
            FULFILLED: (state, action) => {...},
            REJECTED: (state, action) => {...},
        }
    }),

    bar: handleActions({
        fetch: {
            PENDING: (state, action) => {...},
            FULFILLED: (state, action) => {...},
            REJECTED: (state, action) => {...},
        }
    }),

})
```

However until separators are hardcoded I can't do that. Instead I have to change reducer to following:
```js
const reducer = combineReducers({

    foo: handleActions({
        fetch_PENDING: (state, action) => {...},
        fetch_FULFILLED: (state, action) => {...},
        fetch_REJECTED: (state, action) => {...},
    }),

    bar: handleActions({
        fetch_PENDING: (state, action) => {...},
        fetch_FULFILLED: (state, action) => {...},
        fetch_REJECTED: (state, action) => {...},
    }),

})
```

**Comment by tomatau** [comment]
If suggest using explicit coupling between action creators and reducers based on constants rather than implicit coupling based on object shape and library magic. But I can understand a little benefit to your suggestion. Let's wait for @pburtchaell to share his thoughts on this

**Comment by roboslone** [comment]
@pburtchaell ping

**Comment by pburtchaell** [comment]
I agree with @tomatau that explicit coupling of action creators and reducers is better and that it's best to avoid (magical 🦄 ) libraries like Redux Action, but that's my personal preference.

I'd honestly rather not add another config option to the middleware (can potentially bloat the codebase), but I understand your case, @roboslone, and it's a reasonable 3 line diff. I think we can merge this in once the PR is approved. Just add tests, please. 👍

**Comment by pburtchaell** [comment]
Released as `redux-promise-middleware@4.4.1` on npm. Thanks!
